File: Utkarsh mishra/Arrays and String/Questions/singlenumber.java
import java.util.Map.Entry;
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer,Integer> map=new HashMap<>();
       for(int num:nums){
            if(map.containsKey(num))
                map.put(num,map.get(num)+1);
            else
                map.put(num,1);}
       int v=1;

         for(Entry<Integer, Integer> entry: map.entrySet()) {

      if(entry.getValue() == v) {
       return entry.getKey();
    
      
    }}
        
     
         return 0;
} 
}

File: Shreya/Arrays and Strings/Single Number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int,int>mp;
        for(int i=0; i<nums.size(); i++)
        {
            if(mp[nums[i]]==0)
            {
                mp[nums[i]]=1;
            }
            else
            {
                mp[nums[i]]+=1;
            }
        }
        for (auto &el :mp)
        {
            if (el.second==1)
            {
                return el.first;
            }
        }
        return -1;
    }
};